### DISCUSSION TITLE
Tagging spans with values from MDC

### DISCUSSION THREAD

**Discussion by sakariye** [open-telemetry/opentelemetry-java-instrumentation#10734] [Q&A] [answered]
I was wondering if it was possible through an extension or agent instrumentation to grab values from MDC and add them as span attributes. Modifying the user app is not possible at the moment so I'll have to add MDC values to the span through extension jar.

I've tried the following without success:

```
public class MDCInstrumentation implements TypeInstrumentation {

    @Override
    public ElementMatcher<TypeDescription> typeMatcher() {
        return ElementMatchers.named("org.slf4j.MDC");
    }

    @Override
    public void transform(TypeTransformer typeTransformer) {
        typeTransformer.applyAdviceToMethod(namedOneOf("put"),this.getClass().getName() + "$MDCAdvice");

    }

    public static class MDCAdvice {

        @Advice.OnMethodEnter(suppress = Throwable.class)
        public static void onEnter(@Advice.Argument(0) String key, @Advice.Argument(1) String val) {
            Java8BytecodeBridge.currentSpan().setAttribute("my.otel.attrib." + key, val);
        }
    }
}
```

I'm not sure what I'm doing wrong. Some help would be appreciated.

**Comment by zhihaoHappyeveryday** [comment]
Have you solved the problem of programmatic MDC obtaining Span related information values?

**Reply by sakariye** [reply, marked as answer]
I ended up instrumenting the preHandle method of a HandlerInterceptor on method exit. This is where the MDC values are set for my project. The MDC was accessible in the advice class.
